### PR TITLE
chore: move infer_schema and infer_data_type into vortex-dtype

### PR DESCRIPTION
**Pull request by danking**
I want to use these in PyVortex without depending on vortex-datafusion.

### CODE REVIEW

**Review by AdamGS — DISMISSED** [review]

File: vortex-array/src/arrow/dtype.rs [hunk 1]
```diff
@@ -91,3 +104,201 @@ impl FromArrowType<&Field> for DType {
         }
     }
 }
+
+/// Convert a Vortex [struct DType][DType] to an Arrow [Schema].
+///
+/// # Panics
```
Comment (lwwmanning) [inline vortex-array/src/arrow/dtype.rs]: Stale comment

**Review by robert3005 — CHANGES_REQUESTED** [review]
We should not standardise this code but instead come up with a solution. This method is wrong but necessary for datafusion. Not quite sure what the flow is in pyarrow but I would like to avoid this method being api. At most we should copy paste it as an implementation of pyvortex

File: vortex-array/src/arrow/dtype.rs (hunk 1 above)
Comment (danking) [reply vortex-array/src/arrow/dtype.rs]: Removed.

**Comment by danking** [comment]
@robert3005 we now bail on lists

File: vortex-array/src/arrow/dtype.rs [hunk 2]
```diff
@@ -91,3 +104,192 @@ impl FromArrowType<&Field> for DType {
...
+                    field_dt.is_nullable(),
+                )));
+            }
+
+            DataType::Struct(Fields::from(fields))
+        }
+        // There are four kinds of lists: List (32-bit offsets), Large List (64-bit), List View
+        // (32-bit), Large List View (64-bit). We cannot both guarantee zero-copy and commit to an
+        // Arrow dtype because we do not how large our offsets are.
+        DType::List(..) => vortex_bail!("Unsupported dtype: {}", dtype),
```
Comment (robert3005) [inline vortex-array/src/arrow/dtype.rs:172]: For posterity - we will canonicalize to List View. We should figure out what to do about fixed size lists though

**Review by robert3005 — APPROVED** [review]

**Pull request merged**